### PR TITLE
Fix type error when processing satellite data

### PR DESCRIPTION
**Pull request by Ghazal-Mohammadpour**
# Pull Request Summary
This PR addresses a TypeError encountered when processing satellite data from JASON2. The issue was due to performing comparison operations on lists, which need to be supported for the involved data types. 

## Description
While running a script to process satellite altimetry data, we faced a type error that prevented the script from reading data from JASON2. The error message was as follows:
`TypeError: '>=' not supported between instances of 'list' and 'float'
`
This error was triggered during logical operations within a np.where function, specifically when comparing elements of lists (shs, swsp, swdepth, swdistcoast) with float values from the configuration.

### Issue(s) addressed
The solution implemented involves converting the affected lists (shs, swsp, swdepth, swdistcoast) to numpy arrays. By applying this fix, the script now successfully processes data from a variety of satellites without encountering the previously observed type error.

```
    swdepth = np.array(swdepth)
    swdistcoast = np.array(swdistcoast)
    shs = np.array(shs)
    swsp = np.array(swsp)
```

### Testing
The fix was verified by re-running the script with the same dataset that initially triggered the error. Post-fix, the script executed successfully, processing data from multiple satellite sources without any type errors, confirming the effectiveness of the solution.

I request a review from the maintainers to ensure the changes align with project standards.

### CODE REVIEW

File: ww3tools/ProcSat_Altimeter.py [hunk 1]
```diff
@@ -1,80 +1,3 @@
-#!/usr/bin/env python3
```
Comment (JessicaMeixner-NOAA) [inline ww3tools/ProcSat_Altimeter.py]: Why was this deleted?

File: ww3tools/ProcSat_Altimeter.py [hunk 2]
```diff
@@ -83,6 +6,7 @@
 import netCDF4 as nc
 import xarray as xr
 import pyresample
+from mpl_toolkits.basemap import shiftgrid
```
Comment (JessicaMeixner-NOAA) [inline ww3tools/ProcSat_Altimeter.py]: Is there a reason this new module was includeD?

File: ww3tools/ProcSat_Altimeter.py [hunk 3]
```diff
@@ -146,8 +80,13 @@ def along_track(AODN,atime,wconfig):
                     shscal = np.append(shscal,np.nanmean(AODN['HS_CAL'][indt[indseg]]))
                     swsp = np.append(swsp,np.nanmean(AODN['WSPD'][indt[indseg]]))
                     swspcal = np.append(swspcal,np.nanmean(AODN['WSPD_CAL'][indt[indseg]]))
+
                     # print(repr(t)+" "+repr(i))
                     i=i+1
+    swdepth = np.array(swdepth)
```
Comment (JessicaMeixner-NOAA) [inline ww3tools/ProcSat_Altimeter.py:151]: Is it okay to do this for every satelite or should there be a "if JASON2" statement here?

File: ww3tools/ProcSat_Altimeter.py [hunk 4]
```diff
@@ -347,13 +286,13 @@ def savesat(AODN,wconfig,altsel):
         smethod="AlongTrack"
         hmsg="AODN Altimeter data processed along track."
 
-    datein = datetime.utcfromtimestamp(AODN['TIME'].iloc[0]).strftime('%Y%m%d%H')
-    datefin = datetime.utcfromtimestamp(AODN['TIME'].iloc[-1]).strftime('%Y%m%d%H')
+    datein = datetime.utcfromtimestamp(AODN_ALONGTRACK['TIME'][0]).strftime('%Y%m%d%H')
```
Comment (JessicaMeixner-NOAA) [inline ww3tools/ProcSat_Altimeter.py]: I don't see anything about this update in the text of the PR, can you tell me a little about why we changed from AODN to AODN_ALONGTRACK?

**Review by JessicaMeixner-NOAA — CHANGES_REQUESTED** [review]
I have a few questions that I've left in the code and would like to hear a reason why a large section of the script describing the code was deleted or request that this be added back.   After this is sorted out I will test the code before doing another round of reviews.

File: ww3tools/ProcSat_Altimeter.py (hunk 3 above)
Comment (Ghazal-Mohammadpour) [reply ww3tools/ProcSat_Altimeter.py:151]: The code was tested with other satellites like JASON3, CRYOSAT2, and others on Hera, and it worked without any problems.

**Comment by JessicaMeixner-NOAA** [comment]
Please add to the PR description that this fixes issue #39

File: ww3tools/ProcSat_Altimeter.py (hunk 3 above)
Comment (JessicaMeixner-NOAA) [reply ww3tools/ProcSat_Altimeter.py:151]: Thank you, I will now independently do a test and will report back.

**Comment by JessicaMeixner-NOAA** [comment]
@Ghazal-Mohammadpour I found this issues when trying to run JASON2 for a period that I did not realize there was no JASON2 data.  So I tried again to run for a period of time in which there was no JASON2 data and this time I got a different error: 

For: 
`python ProcSat_Altimeter.py JASON2 2020091400 2020091500 60.0
`

```
wread.aodn_altimeter  successfully completed in 429 seconds. JASON2
Traceback (most recent call last):
  File "/work2/noaa/marine/jmeixner/TestPR43/ww3-tools/ww3tools/ProcSat_Altimeter.py", line 435, in <module>
    savesat(AODN_ALONGTRACK,wconfig,altsel)
  File "/work2/noaa/marine/jmeixner/TestPR43/ww3-tools/ww3tools/ProcSat_Altimeter.py", line 354, in savesat
    datein = datetime.utcfromtimestamp(AODN['TIME'].iloc[0]).strftime('%Y%m%d%H')
  File "/work2/noaa/marine/jmeixner/general/pythonenv/ww3tools/lib/python3.10/site-packages/pandas/core/indexing.py", line 1153, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/work2/noaa/marine/jmeixner/general/pythonenv/ww3tools/lib/python3.10/site-packages/pandas/core/indexing.py", line 1714, in _getitem_axis
    self._validate_integer(key, axis)
  File "/work2/noaa/marine/jmeixner/general/pythonenv/ww3tools/lib/python3.10/site-packages/pandas/core/indexing.py", line 1647, in _validate_integer
    raise IndexError("single positional indexer is out-of-bounds")
IndexError: single positional indexer is out-of-bounds
```

I do not get an error when I choose a time period of which JASON2 has satelite info, for example: 
python ProcSat_Altimeter.py JASON2 2018091400 2018091500 60.0

If others agree we can merge this code, but then I think we need to also have a more graceful exit to the code if there is no data.   Do you also get the error I'm seeing above for a period in which we do not have JASON2 data or is there something on orion/my environment?

**Review by JessicaMeixner-NOAA — APPROVED** [review]
This solves the underlying issues.  Thanks @Ghazal-Mohammadpour

**Comment by JessicaMeixner-NOAA** [comment]
@Ghazal-Mohammadpour  Can you share what a good commit message for this PR will be? As soon as we have that I will merge.

**Comment by Ghazal-Mohammadpour** [comment]
Updates for JASON2 and to exit with message when there is no data
